lib/auth/oidc.go

```diff
@@ -708,6 +713,27 @@ func (a *Server) createOIDCUser(p *createUserParams, dryRun bool) (types.User, e
 	return user, nil
 }
 
+// usernameFromClaims gets the username of the OIDC user based on the claims received. The `preferred_username` field in the OIDC
+// config specifies which claim from the OIDC provider to use as the user's username. If it isn't specified, the email will be used
+// as the username. If it is specified but specifies a claim that doesn't exist, an error is returned.
+func usernameFromClaims(connector types.OIDCConnector, claims jose.Claims, ident *oidc.Identity) (string, error) {
+	usernameClaim := connector.GetUsernameClaim()
+
+	if usernameClaim == "" {
+		return ident.Email, nil
+	}
+
+	username, ok, err := claims.StringClaim(usernameClaim)
+
```

Comment:
Remove empty line here, since the code that follows is directly related to this line.